Hw1 fix

Awek-015 opened the pull request:
## 📌 Summary
<!-- Describe the purpose of this PR and what it does. -->

## 🔍 Changes
- [x] Feature Implementation
- [x] Bug Fix
- [x] Documentation Update
- [x] CI/CD Improvements
- [x] Code Refactoring
- [x] Other (Specify)

## ✅ Checklist
- [x] My code follows the repository's coding style.
- [x] I have run `ruff check .` to ensure code formatting and linting.
- [x] I have run `mypy src/` to check for type errors.
- [x] I have written unit tests and they pass successfully.
- [x] I have verified the test coverage using `coverage run -m pytest`.
- [x] My changes do not break existing functionality.
- [x] Any necessary documentation updates have been made.

## 🔗 Related Issues
<!-- If this PR resolves an issue, reference it here. Example: Closes #12 -->

## 🚀 How to Test
1. **Run static analysis**:
   ```sh
   ruff check . && mypy src/

Awek-015 committed:
Set up CircleCI pipeline

Awek-015 committed:
Merge pull request #2 from Awek-015/feature-ci

Set up CircleCI pipeline

Awek-015 committed:
test

Awek-015 committed:
test

Awek-015 committed:
test2

Awek-015 committed:
test3

Awek-015 committed:
Merge pull request #3 from Awek-015/feature-ci

Feature ci

Awek-015 committed:
Update issue templates

Awek-015 committed:
Merge pull request #4 from Awek-015/Awek-015-patch-1

Update issue templates

Awek-015 committed:
fix

Awek-015 committed:
Merge pull request #5 from Awek-015/fix

fix

Awek-015 committed:
fix2

Awek-015 committed:
Merge branch 'main' into fix

Awek-015 committed:
Merge pull request #6 from Awek-015/fix

fix2

Awek-015 committed:
test

Awek-015 committed:
test

Awek-015 committed:
Merge pull request #7 from Awek-015/fix

Fix

Awek-015 committed:
f

Awek-015 committed:
Merge pull request #8 from Awek-015/fix

f

Awek-015 committed:
cicd update

Awek-015 committed:
Merge branch 'main' into fix

Awek-015 committed:
Merge pull request #9 from Awek-015/fix

cicd update

Awek-015 committed:
fix

Awek-015 committed:
fix

Awek-015 committed:
1

Awek-015 committed:
2

Awek-015 committed:
Merge pull request #10 from Awek-015/fix2

fix

Awek-015 committed:
update ci/cd

Awek-015 committed:
Merge branch 'main' into fix2

Awek-015 committed:
Merge pull request #11 from Awek-015/fix2

update ci/cd

Awek-015 committed:
update docs

Awek-015 committed:
Merge branch 'fix2' of github.com:Awek-015/OSPSD_HW1 into fix2

Awek-015 committed:
Merge pull request #12 from Awek-015/fix2

Fix2

XuanyuYang committed:
feat: Add component Calculator and unit tests

XuanyuYang committed:
fix: Fixed static analysis issues

XuanyuYang committed:
fix: Fixed un-formatted import block...

Awek-015 committed:
Merge pull request #13 from Awek-015/feature-calculator

jasmineyuan169 committed:
Developed logger component and added unit tests

jasmineyuan169 committed:
Developed logger component and added unit tests

jasmineyuan169 committed:
Developed logger component and added unit tests

jasmineyuan169 committed:
Developed logger component and added unit tests

jasmineyuan169 committed:
Developed logger component and added unit tests

Awek-015 committed:
Merge pull request #14 from Awek-015/logger-dev

Logger dev

guanroger committed:
adding notifier and notifier unit test

guanroger committed:
fixxing notifier bug

guanroger committed:
fixxing notifier bug

guanroger committed:
fixxing notifier bug

guanroger committed:
fixxing notifier bug

XuanyuYang committed:
update readme

guanroger committed:
fixxing notifier bug

guanroger committed:
fixxing notifier bug

Awek-015 committed:
Merge pull request #17 from Awek-015/notifier-dev

Notifier dev

Awek-015 committed:
Merge branch 'main' into update-readme

Awek-015 committed:
Merge pull request #18 from Awek-015/update-readme

update readme

Awek-015 committed:
Update README.md

Awek-015 committed:
Merge pull request #19 from Awek-015/Awek-015-patch-1

Update README.md

BEACON13 committed:
Add integration test

BEACON13 committed:
Add end-to-end test

BEACON13 committed:
fix tests

BEACON13 committed:
Merge branch 'main' into tests

Awek-015 committed:
Merge pull request #20 from Awek-015/tests

Awek-015 committed:
update docs

Awek-015 committed:
Merge pull request #21 from Awek-015/updatedocs

update docs

XuanyuYang committed:
Updated some links

XuanyuYang committed:
Update README.md

Awek-015 committed:
Merge pull request #22 from Awek-015/udpate-readme

Udpate readme

Awek-015 committed:
improvement

Awek-015 committed:
improvement

Awek-015 committed:
修复项目构建配置和 CircleCI 配置

Awek-015 committed:
fix

Awek-015 committed:
fix

Awek-015 committed:
fix

Awek-015 committed:
fix

Awek-015 committed:
fix

Awek-015 committed:
fix

Awek-015 committed:
Merge pull request #23 from Awek-015/new-single-commit

New single commit

Awek-015 committed:
Remove duplicate unit tests and update documentation

Awek-015 committed:
update

Awek-015 committed:
Merge pull request #24 from Awek-015/new-single-commit

New single commit

Davina0316 commented:
Excellent job! Perhaps requirement.txt is not needed because uv installs everything? Still a great job!

Rachel-ty commented on docs/component.md:
well done!

Rachel-ty commented on requirements.txt:
uv project does not need requirements.txt

Rachel-ty replied:
use pyproject.toml to manage dependency instead

Rachel-ty commented on src/logger/tests/test_logger.py:
```diff
@@ -0,0 +1,46 @@
+        """Test error logging functionality."""
+        logger = create_logger()
+        logger.log_error("Sample error")
+        mocked_log.assert_called_once_with("Error: Sample error")
```
why use mock in unit test?

Wiliiam-RWL commented on .circleci/config.yml:
```diff
@@ -1,28 +1,141 @@
+            . .venv/bin/activate
+            uv pip install pytest pytest-cov mypy ruff coverage
+            uv pip install -e .
+      - run:
+          name: Run static analysis (ruff & mypy)
+          command: |
+            . .venv/bin/activate
+            ruff check .
+            mypy src/
```
Static analysis is not a concept under unit testing, so it might be better to move this step out of the job of unit test.

Wiliiam-RWL commented on .circleci/config.yml:
```diff
@@ -1,28 +1,141 @@
-          command: pip install pytest
+          name: Install dependencies
+          command: |
+            pip install uv
+            uv venv
+            . .venv/bin/activate
+            uv pip install pytest pytest-cov mypy ruff coverage
+            uv pip install -e .
```
It seems that you are installing uv and the same dependencies for every job. Consider using 'persist_to_workspace' to avoid repeated environment set up. 😊

unknownblueguy6 commented:
TA Feedback:
You have received good feedback from your peers, do incorporate that as well.
You should use the install script to install uv rather than use pip.
Unit tests are not passing, make sure to fix that. Static analysis and formatting should be its own step, rather than being a part of the unit tests.
Component structure is good! Refer to the Python example in this repo as well: https://github.com/nyuoss/mAIgic-nyu/tree/ky/oss-2

XuanyuYang committed:
Fix: Update UV dependencies, fix unit tests, refactor CI pipeline

- Updated UV dependencies configuration, removed requirements.txt
- Moved static analysis (ruff & mypy) to a separate step
- Fixed issues causing unit tests to fail

XuanyuYang committed:
fix

XuanyuYang committed:
fix

XuanyuYang committed:
Fix: add persist_to_workspace

XuanyuYang committed:
fix typo...

XuanyuYang committed:
fix

XuanyuYang committed:
fix

XuanyuYang committed:
Fix: use script to install uv instead of pip

XuanyuYang committed:
fix

Davina0316 reviewed (commented):
Excellent job! Perhaps requirement.txt is not needed because uv installs everything? Still a great job!
